lib/manager/poetry/artifacts.ts

```diff
@@ -37,6 +39,50 @@ function getPythonConstraint(
   return undefined;
 }
 
+function getPoetrySources(content: string, fileName: string): PoetrySource[] {
+  let pyprojectFile: PoetryFile;
+  try {
+    pyprojectFile = parse(content);
+  } catch (err) {
+    logger.debug({ err }, 'Error parsing pyproject.toml file');
+    return [];
+  }
+  if (!pyprojectFile.tool?.poetry) {
+    logger.debug(`{$fileName} contains no poetry section`);
+    return [];
+  }
+
+  const sources = pyprojectFile.tool?.poetry?.source;
+  const sourceSet = new Set<PoetrySource>();
```

Comment:
Does using a Set here provide any advantage versus regular array? As it contains objects, I don't think we'll get any de-duplication?

Reply:
Oops -- you're right, we won't get dedpulication. Changed it to an array.